frontend/src/app/components/shared/crm-table/crm-table.component.html

```diff
@@ -27,21 +29,20 @@
                 </mat-checkbox>
             </td>
         </ng-container>
-        <ng-container *ngFor="let map of keyDisplayedNameOrdering" [matColumnDef]="map.key">
-            <div *ngIf="map.key.indexOf('Name') > -1; else normalCell">
+        <ng-container *ngFor="let map of keyDisplayedNameMaps" [matColumnDef]="map.key">
+            <div>
                 <th mat-header-cell *matHeaderCellDef>{{ map.displayedName }}</th>
-                <td mat-cell *matCellDef="let element">
-                    <mat-form-field floatLabel="never">
-                        <input matInput (value)="element[map.key]"
+                <td [ngClass]="map.key"  mat-cell *matCellDef="let element"> 
+                    <mat-form-field floatLabel="never" *ngIf="map.displayedName.indexOf('Name') > -1; else normalCell">
+                        <input class="editable" *ngIf="map.displayedName.indexOf('Name') > -1" matInput [value]="element[map.key]"
                             [(ngModel)]="element[map.key]"
                             (change)="emitRename(element)">
                     </mat-form-field>
+                    <ng-template #normalCell>
+                        {{ element[map.key] }}
+                    </ng-template>
```

Comment:
Do we need to check ngIf for both the mat-form-field and the inner input tag? Wouldn't the ngIf in the mat-form-field guarantee that the ngIf in the inner input is always true (since we already checked it outside?)

Reply:
Yes, good catch.